tests/components/netatmo/test_climate.py

```diff
@@ -0,0 +1,539 @@
+"""The tests for the Netatmo climate platform."""
+from unittest.mock import Mock
+
+import pytest
+
+from homeassistant.components.climate import (
+    DOMAIN as CLIMATE_DOMAIN,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_PRESET_MODE,
+    SERVICE_SET_TEMPERATURE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+)
+from homeassistant.components.climate.const import (
+    ATTR_HVAC_MODE,
+    ATTR_PRESET_MODE,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    PRESET_AWAY,
+    PRESET_BOOST,
+)
+from homeassistant.components.netatmo import climate
+from homeassistant.components.netatmo.climate import (
+    NA_THERM,
+    NA_VALVE,
+    PRESET_FROST_GUARD,
+    PRESET_SCHEDULE,
+)
+from homeassistant.components.netatmo.const import (
+    ATTR_SCHEDULE_NAME,
+    SERVICE_SET_SCHEDULE,
+)
+from homeassistant.components.webhook import async_handle_webhook
+from homeassistant.const import ATTR_ENTITY_ID, ATTR_TEMPERATURE, CONF_WEBHOOK_ID
+from homeassistant.util.aiohttp import MockRequest
+
+
+async def simulate_webhook(hass, webhook_id, response):
+    """Simulate a webhook event."""
+    request = MockRequest(content=response, mock_source="test")
+    await async_handle_webhook(hass, webhook_id, request)
+    await hass.async_block_till_done()
+
+
+async def test_webhook_event_handling_thermostats(hass, climate_entry):
+    """Test service and webhook event handling with thermostats."""
+    webhook_id = climate_entry.data[CONF_WEBHOOK_ID]
+    climate_entity_livingroom = "climate.netatmo_livingroom"
+
+    assert hass.states.get(climate_entity_livingroom).state == "auto"
+    assert (
+        hass.states.get(climate_entity_livingroom).attributes["preset_mode"]
+        == "Schedule"
+    )
+    assert hass.states.get(climate_entity_livingroom).attributes["temperature"] == 12
+
+    # Test service setting the temperature
+    await hass.services.async_call(
+        CLIMATE_DOMAIN,
+        SERVICE_SET_TEMPERATURE,
+        {ATTR_ENTITY_ID: climate_entity_livingroom, ATTR_TEMPERATURE: 21},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    # Fake webhook thermostat manual set point
+    response = (
+        b'{"user_id": "91763b24c43d3e344f424e8b", "user": {"id": "91763b24c43d3e344f424e8b",'
```

Comment:
I'm not sure what's most convenient to work with, but if we want to define a dictionary instead for the mock response we could let the webhook simulate helper serialize the data to a byte string.

Reply:
I don't think it matters, but good idea though.